x-pack/plugins/apm/public/components/shared/Links/apm/AnomalyDetectionSetupLink.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React from 'react';
+import { EuiButtonEmpty, EuiToolTip, EuiIcon } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { APMLink } from './APMLink';
+import { getEnvironmentLabel } from '../../../../../common/environment_filter_values';
+import { useUrlParams } from '../../../../hooks/useUrlParams';
+import { useFetcher, FETCH_STATUS } from '../../../../hooks/useFetcher';
+
+export function AnomalyDetectionSetupLink() {
+  const { uiFilters } = useUrlParams();
+  // check both uiFilters and urlParams for selected environment
+  const environment = uiFilters.environment;
+
+  const { data = { jobs: [], hasLegacyJobs: false }, status } = useFetcher(
+    (callApmApi) =>
+      callApmApi({ pathname: `/api/apm/settings/anomaly-detection` }),
+    [],
+    { preservePreviousData: false }
+  );
+  const isFetchSuccess = status === FETCH_STATUS.SUCCESS;
+
+  // Show alert if there are no jobs OR if no job matches the current environment
+  const showAlert =
+    isFetchSuccess && !data.jobs.some((job) => environment === job.environment);
+  const toolTipText = environment
+    ? getNotEnabledForEnvironmentText(environment)
+    : NOT_ENABLED_TEXT;
+
+  return (
+    <APMLink path="/settings/anomaly-detection">
+      <EuiButtonEmpty size="s" color="primary" iconType="inspect">
+        {ANOMALY_DETECTION_LINK_LABEL}
+      </EuiButtonEmpty>
+      {showAlert && (
+        <EuiToolTip position="bottom" content={toolTipText}>
```

Comment:
tiny nit: colocating the tooltip text logic will make it a little cleaner

```suggestion
        <EuiToolTip position="bottom" content={getToolTipText(environment)}>
```

```ts
function getTooltipText(environment?: string) {
  if (!environment) {
    return i18n.translate("xpack.apm.anomalyDetectionSetup.notEnabledText", {
      defaultMessage: `Anomaly detection is not yet enabled. Click to continue setup.`,
    });
  }

  return i18n.translate(
    "xpack.apm.anomalyDetectionSetup.notEnabledForEnvironmentText",
    {
      defaultMessage: `Anomaly detection is not yet enabled for the "{currentEnvironment}" environment. Click to continue setup.`,
      values: { currentEnvironment: getEnvironmentLabel(environment) },
    }
  );
}
```